samples/app-anonymous-users/csharp/Bots/AnonymousUsersBot.cs

```diff
@@ -0,0 +1,285 @@
+        {
+            int usersCount = 0;
+            int anonymousUsersCount = 0;
+            bool sendAnonymousCount = false;
+            var teamsChannelId = turnContext.Activity.TeamsGetChannelId();
+            var serviceUrl = turnContext.Activity.ServiceUrl;
+            var credentials = new MicrosoftAppCredentials(_appId, _appPassword);
+            ConversationReference conversationReference = null;
+
+            var members = await GetPagedMembers(turnContext, cancellationToken);
+
+            foreach (var teamMember in members)
+            {
+                var proactiveMessage = MessageFactory.Text($"Hello {teamMember.GivenName} {teamMember.Surname}. I'm a Teams conversation bot that support anonymous users.");
+
+                if (teamMember.UserRole != null && teamMember.UserRole == "user")
+                {
+                    usersCount++;
+                }
+
+                if (teamMember.UserRole != null && teamMember.UserRole == "anonymous")
+                {
+                    anonymousUsersCount++;
+                }
+
+                var conversationParameters = new ConversationParameters
+                {
+                    IsGroup = false,
+                    Bot = turnContext.Activity.Recipient,
+                    Members = new ChannelAccount[] { teamMember },
+                    TenantId = turnContext.Activity.Conversation.TenantId,
+                };
+
+                try
+                {
+                    // Creates a conversation on the specified groupchat and send file consent card on that conversation.
+                    await ((CloudAdapter)turnContext.Adapter).CreateConversationAsync(
+                        credentials.MicrosoftAppId,
+                        teamsChannelId,
+                        serviceUrl,
+                        credentials.OAuthScope,
+                        conversationParameters,
+                        async (turnsContextActiT1, cancelTokenC1) =>
+                        {
+                            conversationReference = turnsContextActiT1.Activity.GetConversationReference();
+                            await ((CloudAdapter)turnContext.Adapter).ContinueConversationAsync(
+                                _appId,
+                                conversationReference,
+                                async (turnsContextActiT2, cancelTokenC2) =>
+                                {
+                                    await turnsContextActiT2.SendActivityAsync(proactiveMessage, cancelTokenC2);
+                                },
+                                cancellationToken);
+                        },
+                        cancellationToken);
+                }
+                catch (Exception exception)
+                {
+                    _logger.LogError("Error:" + exception.Message);
+                    sendAnonymousCount = true;
```

Comment:
Add a condition to check if error is occurred for anonymous user and set the flag  sendAnonymousCount = true;

Reply:
Add comments that this condition is handling error for anonymous users because "bot can not creation........

Reply:
Fixed